docs/concepts/tx-lifecycle.md

```diff
@@ -0,0 +1,130 @@
+		|   		      |		           v          		|                     |
+		|   		      |		-----------------------		|                     |
+		|   		      |		| DeliverTxState(t)(1)|		|                     |
+		|   		      |		-----------------------		|                     |
+DeliverTx()	|   		      |	                   |			|                     |
+		|   		      |			   v			|                     |
+		|   		      |		-----------------------		|                     |
+		|   		      |	      	| DeliverTxState(t)(2)|		|                     |
+		|   		      |		-----------------------		|                     |
+DeliverTx()	|   		      |	                   |			|                     |
+		|   		      |			   v			|                     |
+		|   		      |		-----------------------		|                     |
+		|   		      |	      	| DeliverTxState(t)(3)|		|                     |
+		-----------------------		-----------------------		-----------------------
+Commit()		  |				   |				   |
+			  v				   v				   v
+		-----------------------		-----------------------		-----------------------
+States synced:  | CheckTxState(t+1)   |         | DeliverTxState(t+1) |		|   QueryState(t+1)   |
+		-----------------------		|                     |		|		      |
+		          |			|                     |		|		      |
+			  v			|                     |		|	  	      |
+			  .				   .				   .
+			  .				   . 				   .
+			  .				   .  				   .
+			  
+```
+### PostChecks
+`CheckTx` also returns `GasUsed` which may or may not be less than the user's provided `GasWanted`. A `PostCheckFunc` is an optional filter run after `CheckTx` that can be used to enforce that users provide enough gas. 
+
+
+## Consensus
+At each round, a proposer is chosen amongst the validators to create and propose the next block. This validator (presumably honest) has generated a Mempool of validated transactions and now includes them in a block. The validators execute [Tendermint BFT Consensus](https://tendermint.com/docs/spec/consensus/consensus.html#terms); with 2/3 approval from the validators, the block is committed. Note that not all blocks have the same number of transactions and it is possible for consensus to result in a nil block or one with no transactions - here, it is assumed that the transaction has made it this far. 
+
+## State Changes
+During consensus, the validators came to agreement on not only which transactions but also the precise order of the transactions. However, apart from committing to this block in consensus, the ultimate goal is actually for nodes to commit to a new state generated by the transaction state changes. 
+In order to execute the transaction, the following ABCI function calls are made in order to communicate to the application what state changes need to be made. While nodes each run everything individually, since the messages' state transitions are deterministic and the order was finalized during consensus, this process yields a single, unambiguous result.
```

Comment:
Where? How? 

Where: On each full-node (not necessarily validator nodes). 

How: Each full-node receives block candidates and validator signatures. The commit happens locally. All this is handled by Tendermint. It's important to clarify that, as readers might understand that full-node receive block AFTER they have been signed, whereas all this happens simultaneously (everybody, including non validator nodes, receive prevotes and precommits)